Fix outdated repo docs (#4995)
stacks-network/stacks-core: stacks-signer/release-process.md

Reviewer: General note about this: I understand why it is here, it was a little confusing at first (mentions to "this repository", "this project" might better be scoped with `stacks-signer`). 

Also it repeats most of the info in `docs/release-process.md` with slightly differences. Not sure if there's a way to make it DRYer (or if it is worth doing so).

Author: yes - i'm still in the dark a bit about how the signer release process *should* work. i'm relying on @jferrant and @hstove to help correct that document. 

what you see is a best guess using what we have already for stacks-core. i fully expect it will be updated with more feedback.

Reviewer 2: I have no opinion on this process. It does seem a bit confusing but also not sure what could be done to make it any less confusing LOL

The only thing that might be slightly helpful is if there is anything we could add to this doc about testing the release is functioning as expected.

Author: thanks! could you share what is confusing so i can try to button it up?
i've used the existing stacks-core release doc as a template here (indeed, it's mostly just copy/pasted with some minor adjustments). 

regarding the testing - that's where i could use some help. for example, with stacks-node we will do genesis sync etc to verify the binary is working as expected, but i'm not sure if there's anything similar for the signer we could do - if you have any ideas, i'd love to add them to that section.

| Linux ARMv7                 | _builds are provided but not tested_ |
| Linux ARM64                 | _builds are provided but not tested_ |

## Release Schedule and Hotfixes

`stack-signer` releases that add new or updated features shall be released in an ad-hoc manner.
It is generally safe to run a `stacks-signer` from that branch, though it has received less rigorous testing than release branches.
If bugs are found in the `develop` branch, please do [report them as issues](https://github.com/stacks-network/stacks-core/issues) on this repository.

For fixes that impact the correct functioning or liveness of the signer, _hotfixes_ may be issued. These hotfixes are categorized by priority
according to the following rubric:

- **High Priority**. Any fix for an issue that could deny service to the network as a whole, e.g., an issue where a particular kind of invalid transaction would cause nodes to stop processing requests or shut down unintentionally.
- **Medium Priority**. Any fix for an issue that could deny service to individual nodes.
- **Low Priority**. Any fix for an issue that is not high or medium priority.

## Versioning

This project uses a 6 part version number.

> When there is a stacks-core release, `stacks-signer` will assume the same version as the tagged `stacks-core` release ([5 part version](../docs/release-process.md#versioning)).

Reviewer 3: We're not quite following this today, right? E.g. stacks-core is at 2.5.0.0.6, but signer releases are still 2.5.0.0.5.x

Author: correct - the reason is that the signer's code is still in develop branch, but a lot of the `stacks-node` code in develop was not ready for release. post 3.0, the above should be true.

Author: to close this loop - we did discuss bringing over the stacks-signer changes out of the develop branch, but in practice it would have been incredibly messy/difficult to accomplish without a lot of time for testing to ensure we pulled in all of the changes.

| Linux ARMv7                 | _builds are provided but not tested_ |
| Linux ARM64                 | _builds are provided but not tested_ |

## Release Schedule and Hotfixes

> Normal releases in this repository that add new or updated features shall be released in an ad-hoc manner.
> The currently staged changes for such releases are in the [develop branch](https://github.com/stacks-network/stacks-core/tree/develop).

Reviewer: These lines confused me because `docs/release-process` mention a monthly release schedule.

Also, "this repository" here refers to both `stacks-core` and `stacks-signer`. Maybe just:

```suggestion
`stack-signer` releases that add new or updated features shall be released in an ad-hoc manner.
```
